fe/fe-core/src/main/java/com/starrocks/common/util/SqlCredentialRedactor.java

```diff
@@ -0,0 +1,100 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.common.util;
+
+import com.google.common.collect.ImmutableSet;
+import com.starrocks.connector.share.credential.CloudConfigurationConstants;
+
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class to redact sensitive credentials from SQL strings.
+ * This is primarily used when SQL parsing fails and we need to log the original SQL statement
+ * without exposing credentials in audit logs.
+ */
+public class SqlCredentialRedactor {
+
+    // Set of credential keys that should be redacted
+    private static final Set<String> CREDENTIAL_KEYS = ImmutableSet.<String>builder()
+            .add(CloudConfigurationConstants.AWS_S3_ACCESS_KEY)
+            .add(CloudConfigurationConstants.AWS_S3_SECRET_KEY)
+            .add(CloudConfigurationConstants.AWS_S3_SESSION_TOKEN)
+            .add(CloudConfigurationConstants.AWS_GLUE_ACCESS_KEY)
+            .add(CloudConfigurationConstants.AWS_GLUE_SECRET_KEY)
+            .add(CloudConfigurationConstants.AWS_GLUE_SESSION_TOKEN)
+            .add(CloudConfigurationConstants.AZURE_BLOB_SHARED_KEY)
+            .add(CloudConfigurationConstants.AZURE_BLOB_SAS_TOKEN)
+            .add(CloudConfigurationConstants.AZURE_ADLS1_OAUTH2_CREDENTIAL)
+            .add(CloudConfigurationConstants.AZURE_ADLS2_SHARED_KEY)
+            .add(CloudConfigurationConstants.AZURE_ADLS2_SAS_TOKEN)
+            .add(CloudConfigurationConstants.AZURE_ADLS2_OAUTH2_CLIENT_SECRET)
+            .add(CloudConfigurationConstants.GCP_GCS_SERVICE_ACCOUNT_PRIVATE_KEY)
+            .add(CloudConfigurationConstants.GCP_GCS_SERVICE_ACCOUNT_PRIVATE_KEY_ID)
+            .add(CloudConfigurationConstants.HDFS_PASSWORD)
+            .add(CloudConfigurationConstants.HDFS_PASSWORD_DEPRECATED)
+            .add(CloudConfigurationConstants.HDFS_KERBEROS_KEYTAB_DEPRECATED)
+            .add(CloudConfigurationConstants.HADOOP_KERBEROS_KEYTAB)
+            .add(CloudConfigurationConstants.HDFS_KERBEROS_KEYTAB_CONTENT_DEPRECATED)
+            .add(CloudConfigurationConstants.HADOOP_KERBEROS_KEYTAB_CONTENT)
+            .add(CloudConfigurationConstants.ALIYUN_OSS_ACCESS_KEY)
+            .add(CloudConfigurationConstants.ALIYUN_OSS_SECRET_KEY)
+            .add(CloudConfigurationConstants.TENCENT_COS_ACCESS_KEY)
+            .add(CloudConfigurationConstants.TENCENT_COS_SECRET_KEY)
+            .add("password")
+            .add("passwd")
+            .add("pwd")
+            .build();
+
+    // Pattern to match key-value pairs in SQL
+    // This pattern handles cases like:
+    // "key" = "value"
```

Comment:
Is single quote supported? like `'key' = 'value'`.

Reply:
good suggestion. I have fixed it.